edb/wasm/wasm_ext.py

```diff
@@ -0,0 +1,116 @@
+import asyncio
+import http
+import json
+import logging
+import pickle
+
+from functools import partial
+
+from edb import errors
+from edb.server.protocol import protocol
+from edb.server.dbview import dbview
+
+
+log = logging.getLogger(__name__)
+
+
+class WasmRpcError(Exception):
+    def __init__(self, info):
+        super().__init__("Wasm request error")
+        self.__dict__.update(info)
+
+
+def handle_error(
+    request: protocol.HttpRequest,
+    response: protocol.HttpResponse,
+    error: Exception,
+):
+    er_type = type(error)
+    if not issubclass(er_type, errors.EdgeDBError):
+        er_type = errors.InternalServerError
+
+    # TODO(tailhook) figure out whether we want to expose error message here
+    # for arbitrary errors
+    response.body = json.dumps({
+        'kind': 'error',
+        'error': {
+            'message': str(error),
+            'type': er_type.__name__,
+        }
+    }).encode()
+    response.content_type = b'application/json'
+    response.status = http.HTTPStatus.INTERNAL_SERVER_ERROR
+    response.close_connection = True
+
+
+async def rpc_request(  # type: ignore  # match statement
+    server,  # : server.Server,  # circular import
```

Comment:
Use `if typing.TYPE_CHECKING:` to guard imports for type annotations.

Reply:
```python
if typing.TYPE_CHECKING:
    from edb.server import server
```
Yields this error:
```
    edb/server/protocol/protocol.pyx, line 36, in init edb.server.protocol.protocol
        > from edb.wasm import wasm_ext
    /work/edb/wasm/wasm_ext.py, line 52, in <module>
        > server: server.Server,

NameError: name 'server' is not defined
```

This doesn't help. Is this because of import from cython?

Reply:
Ah, right. Forgot about that :(